src/hipscat_import/catalog/map_reduce.py

```diff
@@ -50,19 +51,30 @@ def _iterate_input_file(
 
     for chunk_number, data in enumerate(file_reader.read(input_file, read_columns=read_columns)):
         if use_hipscat_index:
-            if data.index.name == HIPSCAT_ID_COLUMN:
-                mapped_pixels = hipscat_id_to_healpix(data.index, target_order=highest_order)
+            if isinstance(data, pd.DataFrame):
+                if data.index.name == HIPSCAT_ID_COLUMN:
+                    mapped_pixels = hipscat_id_to_healpix(data.index, target_order=highest_order)
+                else:
+                    mapped_pixels = hipscat_id_to_healpix(data[HIPSCAT_ID_COLUMN], target_order=highest_order)
             else:
                 mapped_pixels = hipscat_id_to_healpix(data[HIPSCAT_ID_COLUMN], target_order=highest_order)
```

Comment:
I think this could be simplified, with one if statement `if isinstance(data, pd.DataFrame) and data.index.name == HIPSCAT_ID_COLUMN`, so there's not the repeated else statements